src/lowering/desugar.ml

```diff
@@ -68,6 +68,18 @@ and exp' at note = function
     I.PrimE (I.RelPrim (!ot, o), [exp e1; exp e2])
   | S.ShowE (ot, e) ->
     I.PrimE (I.ShowPrim !ot, [exp e])
+  | S.ToCandidE es ->
+    (* desugaring to_candid introduces explicit IR tuple: *)
+    let arg = tupE (exps es) in
+    begin match T.normalize arg.note.Note.typ with
+    | T.Tup ts -> (primE (I.SerializePrim [T.Tup ts]) [arg]).it
+    | _ -> assert false
+    end
```

Comment:
```suggestion
    (* desugaring to_candid introduces explicit IR tuple: *)
    let arg = tupE (exps es) in
    begin match T.normalize arg.note.Note.typ with
    | T.Tup ts -> (primE (I.SerializePrim [T.Tup ts]) [arg]).it
    | _ -> assert false
    end
```

Reply:
If you look at the typing of SerializePrim in check_ir.ml, and the way it is used in codegen/compile.ml,
I think you'll come to same conclusion.

Reply:
It took me a few attemps to get this right because I used `tupE args` instead of `seqE args` - the latter matches at the term leve of the T.seq functions used at the typ-level, ()avoiding 1-tuples)

Reply:
https://github.com/dfinity/motoko/pull/3214 is a branch with the necessary change (also suppressing the -v flag in the test)

Reply:
Thanks for digging into this and finding a more direct solution.  I can see how this would work, but the subtle balance of Candid tuples versus "meta-level" Candid value sequences throughout the abstract syntax feels complex and unresolved for me, probably because I'm still not familiar enough, though it's making sense as we go, and this helps a lot.  (Again, Thanks!)

I'll stare at your solution today and try to appreciate more what still feels unresolved for me.